desktop/cmp/treemap/TreeMap.js

```diff
@@ -0,0 +1,317 @@
+import {DarkTheme} from './theme/Dark';
+
+import './TreeMap.scss';
+import {TreeMapModel} from './TreeMapModel';
+
+/**
+ * Component for rendering a TreeMap.
+ *
+ * It is a managed wrapper around a Highcharts TreeMap visualization, which renders
+ * records from a Store and optionally binds to a GridModel.
+ *
+ * @see TreeMapModel
+ */
+@HoistComponent
+@LayoutSupport
+export class TreeMap extends Component {
+
+    static propTypes = {
+        /** Primary component model instance. */
+        model: PT.oneOfType([PT.instanceOf(TreeMapModel), PT.object]).isRequired
+    };
+
+    static modelClass = TreeMapModel;
+
+    baseClassName = 'xh-treemap';
+
+    _chartElem = new Ref();
+    _chart = null;
+
+    constructor(props) {
+        super(props);
+
+        // Detect double-clicks vs single-clicks
+        this._clickCount = 0;
+        this._debouncedClickHandler = debounce(this.clickHandler, 500);
+
+        // Sync selection
+        this.addReaction({
+            track: () => [this.model.selectedIds, this.getMergedConfig()],
+            run: () => this.syncSelection(),
+            delay: 1 // Must wait for chart re-render on data / config change
+        });
+    }
+
+    render() {
+        // Default flex = 1 (flex: 1 1 0) if no dimensions / flex specified, i.e. do not consult child for dimensions;
+        const layoutProps = this.getLayoutProps();
+        if (layoutProps.width == null && layoutProps.height == null && layoutProps.flex == null) {
+            layoutProps.flex = 1;
+        }
+
+        // No-op on first render - will re-render upon setting the _chartElem Ref
+        this.renderHighChart();
+        this.updateLabelVisibilityAsync();
+
+        // Render child item
+        const {data, error} = this.model;
+        let item;
+        if (error) {
+            item = this.renderError(error);
+        } else if (!data.length) {
+            item = this.renderPlaceholder();
+        } else {
+            item = this.renderChartHolder();
+        }
+
+        // Inner div required to be the ref for the chart element
+        return resizeSensor({
+            onResize: debounce((e) => this.resizeChartAsync(e), 100),
+            item: box({
+                ...layoutProps,
+                className: this.getClassName(),
+                item
+            })
+        });
+    }
+
+    renderError(error) {
+        return frame({
+            className: 'xh-treemap__error-message',
+            item: <p>{error}</p>
+        });
+    }
+
+    renderPlaceholder() {
+        return frame({
+            className: 'xh-treemap__placeholder',
+            item: <p>{this.model.emptyText}</p>
+        });
+    }
+
+    renderChartHolder() {
+        return div({
+            className: 'xh-treemap__chart-holder',
+            ref: this._chartElem.ref
+        });
+    }
+
+    //-------------------
+    // Implementation
+    //-------------------
+    renderHighChart() {
+        this.destroyHighChart();
```

Comment:
To ask the question....should we really be destroying the chart on each render cycle? 

I did a bit of experimenting with calling `chart.update()` with the same config (series included) if `_chart` was already defined. That seemed to work and would presumably be more efficient / a better practice if it doesn't have negative side effects...  

Then again, maybe there's no advantage here.  I would be curious to see if there is any broader discussion re. HC and React / render lifecycle.